src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs

```diff
@@ -5,14 +5,14 @@ namespace System.IO.IsolatedStorage
 {
     internal static partial class Helper
     {
-        private const string IsolatedStorageDirectoryName = "IsolatedStorage";
+        public static string IsolatedStorageDirectoryName { get; set;} = "IsolatedStorage";
```

Comment:
I think you can have a Helper.AnyMobile.cs and a Helper.NonMobile.cs that has the old string const behavior and you would not need to have an Initialize method.